cmdstanpy/cmds.py

```diff
@@ -12,32 +12,34 @@
 from multiprocessing import cpu_count
 from multiprocessing.pool import ThreadPool
 from pathlib import Path
-from typing import Dict, List
+from typing import Dict, List, Union, Tuple
 
 from cmdstanpy import TMPDIR
 from cmdstanpy.lib import Model, StanData, RunSet, SamplerArgs
 from cmdstanpy.utils import cmdstan_path
 
 
 def compile_model(
-    stan_file: str = None, opt_lvl: int = 1, overwrite: bool = False,
-    include_paths: List[str] = None
+    stan_file: str = None,
+    opt_lvl: int = 1,
```

Comment:
should the default be 3?

Reply:
that slows down compilation time which is already quite slow.  once we get compilation sped up (Bob's working on that now), we can change default value.

Reply:
Is there a difference in sampling time?

Reply:
a year ago there was some discussion on discourse:  https://discourse.mc-stan.org/t/o2-vs-o3-compiler-optimization-level/3714

we could use boolean arg `optimize` where False is optimization level 1 and True is optimization level either 2 or 3 (sounds like 2 might be preferable)